### PR TITLE
Adds formula for the latest version of the Julius CSR Engine

### PR DESCRIPTION
**Pull request by sexybiggetje**
Contributes the Julius CSR Engine formula.

### CODE REVIEW

File: Library/Formula/julius.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+class Julius < Formula
+  homepage "http://julius.sourceforge.jp"
+  url "http://sourceforge.jp/frs/redir.php?m=iij&f=%2Fjulius%2F60273%2Fjulius-4.3.1.tar.gz"
+  sha1 "88f64ae9ed00b6ab5a2d4fe07e3ced141a46c196"
+
+  depends_on "libsndfile"
+
+  def install
+
```
Comment (mikemcquaid) [inline Library/Formula/julius.rb]: Remove this blank line.

File: Library/Formula/julius.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    system "./configure", "--disable-debug",
+                          "--disable-dependency-tracking",
+                          "--disable-silent-rules",
+                          "--mandir=#{man}",
+                          "--prefix=#{prefix}"
+    system "make", "install"
+  end
+
```
Comment (mikemcquaid) [inline Library/Formula/julius.rb]: Remove this blank line.

File: Library/Formula/julius.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+end
```
Comment (mikemcquaid) [inline Library/Formula/julius.rb]: Can a test be added to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

Comment (sexybiggetje) [inline Library/Formula/julius.rb]: The julius application doesn't really have a easy test method, every command it takes requires at least a language model, which can be obtained from different other projects depending on your language.

--version doesn't ekist, but --help prints out both version info and a long list of switches.

Comment (mikemcquaid) [inline Library/Formula/julius.rb]: `--help` would be better than nothing, thanks.

Comment (sexybiggetje) [inline Library/Formula/julius.rb]: When I add the following test block (tried other examples found in the Formulas) brew test fails. Any clue as how to resolve this?

```ruby
  test do
    result = pipe_output("#{bin}/julius.dSYM --help")
    assert result.include?("Global Options")
  end
```

This is the output it produces:
```bash
martijn@kenpachi-2 /u/local> brew test julius
Testing julius
==> /usr/local/Cellar/julius/4.3.1/bin/julius.dSYM --help
Julius rev.4.3.1 - based on JuliusLib rev.4.3.1 (fast)

Engine specification:
 -  Base setup   : fast
 -  Supported LM : DFA, N-gram, Word
 -  Extension    : LibSndFile
 -  Compiled by  : clang -g -O2

Options:

--- Global Options -----------------------------------------------

** removed most of long help message **

 Additional options for application:
    [--help]	display this help
    [-help]	display this help
    [-outfile]	save result in separate .out file
    [-nolog]	not output any log
    [-logfile arg]	output log to file
    [-separatescore]	output AM and LM scores separately
    [-kanji arg]	convert character set for output
    [-nocharconv]	disable charconv
    [-charconv arg arg]	convert character set for output
    [-outcode arg]	select info to output to the module: WLPSCwlps
    [-module (arg)]	run as a server module
    [-record arg]	record input waveform to file in dir
Error: julius: failed
Failed assertion, no message given.
martijn@kenpachi-2 /u/local> 
```

I tried some other variants to this found in other formula, but even matching the word Julius seems to fail.

Comment (mikemcquaid) [inline Library/Formula/julius.rb]: Does it return a non-zero exit code? (i.e. does `julius --help && echo fish` print `fish`?

Comment (sexybiggetje) [inline Library/Formula/julius.rb]: Apparently it indeed does not print out the word fish. I would assume that julius without params might validly exit > 0, but surely exiting > 0 on printing the help message is a tad bit weird.

Comment (mikemcquaid) [inline Library/Formula/julius.rb]: Use `shell_output` and pass `1` as the second argument.

**Comment by sexybiggetje** [comment]
So I updated to the following:
```ruby
  test do
    result = shell_output("#{bin}/julius.dSYM --help", 1)
    assert result.include?("Global Options")
  end
```

Which produces the exact same output. However, if I say for example require status 2 instead of 1 it produces:
```bash
Error: julius: failed
<2> expected but was
<1>.
```

Which indicates it does exit with status 1 (verified also by executing the command and echo $?).
So it's still stuck at:

```bash
Error: julius: failed
Failed assertion, no message given.
```

**Comment by sexybiggetje** [comment]
Is the debug backtrace on the assertion of any help?

```
/usr/local/Library/Homebrew/debrew.rb:10:in `raise'
MiniTest::Assertion: Failed assertion, no message given.
1. raise
2. ignore
3. backtrace
4. irb
5. shell
Choose an action: 3
/usr/local/Library/Homebrew/debrew.rb:10:in `raise'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/minitest/unit.rb:200:in `assert'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/test/unit/assertions.rb:38:in `assert'
/usr/local/Library/Formula/julius.rb:19:in `block in <class:Julius>'
/usr/local/Library/Homebrew/debrew.rb:29:in `block in test'
/usr/local/Library/Homebrew/debrew.rb:96:in `debrew'
/usr/local/Library/Homebrew/debrew.rb:29:in `test'
/usr/local/Library/Homebrew/formula.rb:609:in `block in run_test'
/usr/local/Library/Homebrew/extend/fileutils.rb:21:in `mktemp'
/usr/local/Library/Homebrew/formula.rb:607:in `run_test'
/usr/local/Library/Homebrew/cmd/test.rb:59:in `block (2 levels) in test'
/System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib/ruby/2.0.0/timeout.rb:66:in `timeout'
/usr/local/Library/Homebrew/cmd/test.rb:58:in `block in test'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `each'
/usr/local/Library/Homebrew/cmd/test.rb:37:in `test'
/usr/local/Library/brew.rb:136:in `<main>'
```

**Comment by mikemcquaid** [comment]
Sorry, status should be `1` which should work.

**Comment by sexybiggetje** [comment]
The problem is that the .include? part fails. The shell_output command fails to capture the data produced by the julius output.

**Comment by jacknagel** [comment]
82fb8c91fd5d9f0d127b02f1d5b988198b8873dd

**Pull request closed without merging**